misk-policy/src/main/kotlin/misk/policy/opa/OpaModule.kt

```diff
@@ -0,0 +1,35 @@
+package misk.policy.opa
+
+import com.google.inject.Provides
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import misk.client.HttpClientFactory
+import misk.inject.KAbstractModule
+import retrofit2.converter.scalars.ScalarsConverterFactory
+import javax.inject.Named
+import javax.inject.Singleton
+
+class OpaModule : KAbstractModule() {
+  override fun configure() {
```

Comment:
optional: instead of injecting the objects below using `@Provider` methods, you could make the configuration object a constructor parameter and then bind these objects in the `configure` method.

That will allow you to also check if the config is empty and then exit early if no config was provided.
It might be useful later, because then you could include this module in the `MiskCommonModule` and not worry about whether configuration was provided on not

Reply:
yea good call, also matches other module patterns.